src/test/java/org/zalando/nakadi/service/EventStreamTest.java

```diff
@@ -343,4 +358,88 @@ public void whenReadFromMultiplePartitionsThenGroupedInBatchesAccordingToPartiti
         assertThat(batches[2], sameJSONAs(jsonBatch("2", "000000000000000000", Optional.of(nCopies(2, DUMMY)))));
     }
 
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testWriteStreamEvent() {
+
+        final Meter meter = mock(Meter.class);
+
+        final EventStream eventStream =
+            new EventStream(null,
+                null,
+                null,
+                null,
+                null,
+                meter,
+                null);
+
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        final Cursor cursor = new Cursor("22", "000000000000000023");
+        final ArrayList<String> events = Lists.newArrayList(
+            "{\"a\":\"b\"}",
+            "{\"c\":\"d\"}",
+            "{\"e\":\"f\"}");
+
+        try {
+            eventStream.writeStreamEvent(baos, cursor, events);
+            final Map<String, Object> batch =
+                mapper.readValue(baos.toString(), new TypeReference<Map<String, Object>>() {});
+
+            final Map<String, String> cursorM = (Map<String, String>) batch.get("cursor");
+            assertEquals("22", cursorM.get("partition"));
+            assertEquals("000000000000000023", cursorM.get("offset"));
+
+            final List<Map<String, String>> eventsM = (List<Map<String, String>>) batch.get("events");
+            assertTrue(eventsM.size() == 3);
+
+            // check the order is preserved as well as the data via get
+            assertEquals("b", eventsM.get(0).get("a"));
+            assertEquals("d", eventsM.get(1).get("c"));
+            assertEquals("f", eventsM.get(2).get("e"));
+
+            verify(meter, times(1)).mark(anyInt());
+
+        } catch (IOException e) {
+            fail(e.getMessage());
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testWriteStreamEventEmpty() {
```

Comment:
@dehora my fault. I was confused about `get` behaviour. It would be clearer for me if the test checked `containsKey("events")` false. But it's fine like this.